### PR TITLE
Add metadata to the triggered bus event

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-docs/issues/516 |
| Dependencies | -- |
| Decisions | As discussed, the event metadata (such as hostname, timestamp) can be inferred by the `RIPE Bus API`. Thus the client passes only something like: ```await busAPI.trigger("topic", {name: "ripe-twitch", origin: "twitch", order: order})``` or something simpler like ```await busAPI.trigger("topic", "hello world") ``` |

### CODE REVIEW

**Comment by joao-conde** [comment]
@gcandal another option would be to include name & origin in the options object currently passed. I like that approach better but currently, options are being used for callbacks only, therefore I opted for this approach. What do you think?

File: js/base.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import * as os from "os";
```
Comment (joamag) [inline js/base.js:1]: this will make the thing dependent on node we should be careful.. we may need to tune rollup.js

**Review by joamag — CHANGES_REQUESTED** [review]

File: js/base.js (hunk 1 above)
Comment (joao-conde) [reply js/base.js:1]: This import was already being used on `kafka-consumer.js` and  `kafka-retry-consumer.js`

Comment (joao-conde) [reply js/base.js:1]: So to change one I'd change all. That should maybe be done in a separate PR?

Comment (joao-conde) [reply js/base.js:1]: Also, kafka consumers are using `fs` aswell.

**Comment by gcandal** [comment]
@joao-conde please check if everything still works (you'll have to adapt the users to the new `trigger` and `bind`).

**Review by joamag — APPROVED** [review]

**Pull request merged**